router/src/server.rs

```diff
@@ -286,6 +292,7 @@ async fn generate(
     }
 
     let details = req.0.parameters.details || req.0.parameters.decoder_input_details;
+    let adapter_id = req.0.parameters.adapter_id.clone();
```

Comment:
Note that following #212, adapter ID can come from the `merged_parameters` as well as the `adapter_id`.

Reply:
good point